src/UI/Implementation/Component/Card/Renderer.php

```diff
@@ -48,6 +62,36 @@ public function render(Component\Component $component, RendererInterface $defaul
 				$tpl->parseCurrentBlock();
 			}
 		}
+
+		if($component instanceof Component\Card\RepositoryObject)
+		{
+			$tpl->setCurrentBlock("action");
+
+			$obj_icon = $component->getObjectIcon();
+			if($obj_icon !== null) {
+				$tpl->setVariable("OBJECT_ICON",$default_renderer->render($obj_icon,$default_renderer));
+			}
+
+			$progress = $component->getProgress();
+			if($progress !== null) {
+				$tpl->setVariable("PROGRESS_STATUS",$default_renderer->render($progress));
+			}
+
+			$certificate = $component->getCertificateIcon();
+			if($certificate !== null) {
+				$certificate_icon= new CustomIcon(ilUtil::getImagePath("icon_cert.svg"), "Certificate", "responsive",false);
```

Comment:
This dep. just occured to me, while looking into the "standard icon" issue. Again, here as well: use the standard icon. Note that this is a highly problematic dep. which you can get rid of this way? This icon here is surely not user uploaded content, right?